packages/blockchain/contracts/interfaces/IAtomicSwap.sol

```diff
@@ -2,76 +2,157 @@
 pragma solidity ^0.8.9;
 
 interface IAtomicSwap {
-    // Data types
+    function onReceivePacket(
```

Comment:
we don't have unfortunately. 
all is just optimistic update before sending cross-chain messaging.

Reply:
hmm, I guess that's good idea, but I'm not sure, there is no gas limit or not. 
currently I'm experiencing gas limit in tx so sometimes it's impossible to update some state variable on counter party chain even. contract to contract call took many gas fee. 
not sure it will work or not.